src/mqt/bench/evaluation/evaluation.py

```diff
@@ -25,97 +17,43 @@
 
 def create_statistics() -> None:
     source_circuits_list = [file for file in Path(utils.get_qasm_output_path()).iterdir() if file.suffix == ".qasm"]
-
-    res = Parallel(n_jobs=-1, verbose=100)(delayed(evaluate_qasm_file)(filename) for filename in source_circuits_list)
+    res_dicts = Parallel(n_jobs=-1, verbose=100)(
+        delayed(evaluate_qasm_file)(str(filename)) for filename in source_circuits_list
+    )
     target_dir = Path(resources.files("mqt.bench") / "evaluation/")
     with Path(target_dir / "evaluation_data.pkl").open("wb") as f:
-        pickle.dump(res, f)
+        pickle.dump(res_dicts, f)
 
 
-def evaluate_qasm_file(filename: str) -> tuple[str, int, int, int, int, float, float, float, float, float]:
+class EvaluationResult(TypedDict):
+    filename: str
+    num_qubits: int
+    depth: int
+    num_gates: int
+    num_multiple_qubit_gates: int
+    program_communication: float
+    critical_depth: float
+    entanglement_ratio: float
+    parallelism: float
+    liveness: float
```

Comment:
Now that I think of it, a `dataclass` is probably more suitable here since the dictionary just creates an unnecessary overhead compared to the `tuple`.

Reply:
Implemented it now as a dataclass